src/viur/core/module.py

```diff
@@ -1,20 +1,201 @@
-from typing import Dict, Any, Union, Tuple, Callable
+import copy
+import inspect
+import typing
+import logging
+from viur.core import errors, current
 from viur.core.config import conf
 
 
+class Method:
+    """
+    Abstraction wrapper for any public available method.
+    """
+
+    @classmethod
+    def ensure(cls, func):
+        """
+        Ensures the provided `func` parameter is either a Method already, or turns it
+        into a Method. This is done to avoid stacking Method objects, which may create
+        unwanted results.
+        """
+        if isinstance(func, Method):
+            return func
+
+        return cls(func)
+
+    def __init__(self, func: typing.Callable):
+        # Attributes
+        self.exposed = None  # None = unexposed, True = exposed, False = internal exposed
+        self.ssl = False
+        self.methods = ("GET", "POST", "HEAD")
+        self.seo_language_map = None
+
+        # Guards
+        self.skey = None
+        self.access = None
+
+        # Content
+        self._func = func
+        self._instance = None
+
+    def __get__(self, obj, objtype):
```

Comment:
[Here](https://docs.python.org/3/howto/descriptor.html) they're using this syntax. Can it be optional?
```suggestion
    def __get__(self, obj, objtype=None):
```

Reply:
Better it is...